plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -630,6 +630,20 @@ def set_update_timestamp
       (template.dig(:metadata, :annotations) || {})[:"samson/updateTimestamp"] = Time.now.utc.iso8601
     end
 
+    def set_well_known_labels
+      return unless KUBERNETES_ADD_WELL_KNOWN_LABELS
+
+      metadata_labels_paths.each do |path|
+        next unless labels = template.dig(*path)
+
+        # always overwrite managed-by label since it is managed by samson
+        labels[:"app.kubernetes.io/managed-by"] = "samson"
+
+        # do not overwrite existing name label, if already set
+        labels[:"app.kubernetes.io/name"] ||= project.name
```

Comment:
I set this to `project.name` as opposed to `project.permalink`. The rationale is that these are not immutable labels and if the project name is changed, so should the label.

However, this will be different from how the `set_project_label` works, where `permalink` is used. Thoughts? Concerns?

Reply:
we use the name for stats, so permalink might be safer ... but 🤷

Reply:
slight preference for permalink just since its usually a nicer tag value and doesn't add more gotchas/duplication